### PR TITLE
Add features to the publishable class

### PR DESCRIPTION
**Pull request by Benolavi**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -7,3 +7,8 @@ gem 'json',  '~> 1.8'
 gem 'bunny', '~> 2.1', '>= 2.1.0'
 gem 'activesupport', '~> 4.2'
 
+gem 'activerecord'
+group :test do
+  gem 'database_cleaner'
+end
+
```
Comment (Benolavi) [inline Gemfile]: Needed for tests

File: lib/videri/publishable.rb [hunk 2]
```diff
@@ -2,63 +2,73 @@
...
-      self.class_name = self.to_s.downcase.pluralize.freeze
+      class_attribute :_topic
+      class_attribute :_class_name
+      class_attribute :_attributes
+      class_attribute :_change_attributes
+
+      # Set defaults
+      self._topic = :resource
+      self._class_name = self.to_s.downcase.pluralize.freeze
+      self._attributes = []
```
Comment (Benolavi) [inline lib/videri/publishable.rb]: The attributes class variable contains the attributes that will be in the payload for create/delete

File: lib/videri/publishable.rb [hunk 3]
```diff
@@ -2,63 +2,73 @@
...
+      self._change_attributes = []
 
+      # Callbacks
       after_create    :publish_create
       after_update    :publish_update
       after_destroy   :publish_destroy
 
       def publish_create
+        attributes = self.as_json.select {|k,_v| _attributes.include?(k.to_sym) }
```
Comment (Benolavi) [inline lib/videri/publishable.rb]: Extract only the required attributes

Comment (ralfkrauklis) [inline lib/videri/publishable.rb]: Skip select if _attributes.empty?

File: lib/videri/publishable.rb (hunk 2 above)
Comment (ralfkrauklis) [inline lib/videri/publishable.rb]: Might be better for the _attributes and _change_attributes are Sets to allow for O(1) lookup and to ensure that they never include duplicates.

File: lib/videri/publishable.rb [hunk 4]
```diff
@@ -2,63 +2,73 @@
...
-          key: MessageBus.key(:models, self.class_name, self.id, :updated),
-          payload: {changes: self.changes, id: self.id}
+          topic: self._topic,
+          key: MessageBus.key(:models, self._class_name, self.id, :updated),
+          payload: { id: self.id, changes: self._changes }
         ) if self.changed?
       end
 
       def publish_destroy
+        attributes = self.as_json.select {|k,_v| _attributes.include?(k.to_sym) }
```
Comment (ralfkrauklis) [inline lib/videri/publishable.rb]: Skip select if _attributes.empty?

**Pull request merged**